django/db/backends/postgresql_psycopg2/client.py

```diff
@@ -7,13 +11,40 @@ class DatabaseClient(BaseDatabaseClient):
     executable_name = 'psql'
 
     def runshell(self):
+        def escape_pgpass(txt):
+            return txt.replace('\\', '\\\\').replace(':', '\\:')
+
         settings_dict = self.connection.settings_dict
         args = [self.executable_name]
-        if settings_dict['USER']:
-            args += ["-U", settings_dict['USER']]
-        if settings_dict['HOST']:
-            args.extend(["-h", settings_dict['HOST']])
-        if settings_dict['PORT']:
-            args.extend(["-p", str(settings_dict['PORT'])])
-        args += [settings_dict['NAME']]
+
+        host = settings_dict.get('HOST', '')
+        port = settings_dict.get('PORT', '')
+        name = settings_dict.get('NAME', '')
+        user = settings_dict.get('USER', '')
+        passwd = settings_dict.get('PASSWORD', '')
+
+        if user:
+            args += ["-U", user]
+        if host:
+            args += ["-h", host]
+        if port:
+            args += ["-p", str(port)]
+        args += [name]
+
+        if passwd:
+            # Create temporary .pgpass file
+            # It's deleted automagically on sys.exit
+            temp_pgpass = tempfile.NamedTemporaryFile(mode='w+')
+            print(
```

Comment:
Actually, I tried write() first, but then I hit a bunch of problems regarding byte/string encoding.
I always use utf-8 on Debian GNU/Linux, but I'm not sure how to proceed so that it works everywhere.
We could use force_byte or whatever, but if the locale is not unicode for exemple iso-8859-15, I'm affraid the utf-8 encoded password will not be accepted by psql that will be runing with default environement.
I tested print() for both python2 & 3, with a 'é' in the password, and it seems ok in all cases.
I'm not sure about windows, I don't have access to one.
After considering all the tests that might be required to have it right, including testing process locale and differences between python 2 & 3, I decided to give up that approach.
print() just works, as far as I can tell. I believe it is less ugly that the alternative...

I could use some help if you think using print is unacceptable, please.